synapse/module_api/__init__.py

```diff
@@ -1202,6 +1208,22 @@ async def defer_to_thread(
         """
         return await defer_to_thread(self._hs.get_reactor(), f, *args, **kwargs)
 
+    async def check_username(self, username: str) -> None:
```

Comment:
Perhaps `check_valid_username` here? (I wasn't sure what was being checked until I read the docstring)

Reply:
Oh, I misread. I thought this was only checking the grammar, but it also checks to see if the username is in use.

Maybe `check_username_available`?

Reply:
Well both are partially misleading imo. `check_username` checks the grammar _and_ the availability.

Reply:
Well put! Let's leave it be.